src/components/Image/index.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import {Image as RNImage} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import lodashGet from 'lodash/get';
+import ONYXKEYS from '../../ONYXKEYS';
+import {defaultProps, propTypes} from './imagePropTypes';
+import RESIZE_MODES from './resizeModes';
+
+class Image extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            imageSource: undefined,
+        };
+    }
+
+    componentDidMount() {
+        this.configureImageSource();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.source === this.props.source) {
+            return;
+        }
+        this.configureImageSource();
+    }
+
+    configureImageSource() {
+        const source = this.props.source;
+        const isAuthTokenRequired = this.props.isAuthTokenRequired;
+        let imageSource = source;
+        if (typeof source !== 'number' && isAuthTokenRequired) {
+            const authToken = lodashGet(this.props, 'session.encryptedAuthToken', null);
+            imageSource = {uri: `${source.uri}?encryptedAuthToken=${encodeURIComponent(authToken)}`};
+        }
+        this.setState({imageSource});
+
+        // If an onLoad callback was specified then manually call it and pass
+        // the natural image dimensions to match the native API
+        if (this.props.onLoad == null) {
+            return;
+        }
+        const uri = typeof imageSource === 'number'
```

Comment:
I see two times we check if `imageSource === 'number'` - maybe we can me this a variable like `imageSourceIsNumber`?

Reply:
I've removed this as its not required for web - its only on native that metro resolves this to a number and not on web https://facebook.github.io/metro/docs/getting-started#assets. I've updated the propTypes to reflect this change too :)